pandas/tests/extension/test_arrow.py

```diff
@@ -1605,3 +1605,207 @@ def test_searchsorted_with_na_raises(data_for_sorting, as_series):
     )
     with pytest.raises(ValueError, match=msg):
         arr.searchsorted(b)
+
```

Comment:
are basically all of our non-IO pyarrow tests in this file?  in principle this should only be for extension-interface stuff, and other array tests would go in tests/arrays/.  This could also reasonably go in tests/series/accessors/.  But if all pyarrow stuff is here then this makes sense.

Reply:
Correct, all the non-IO pyarrow tests are here. I'd be open to refactoring in the future, but some on the non extension interface tests use the extension fixtures